src/backend/distributed/executor/multi_server_executor.c

```diff
@@ -46,70 +46,52 @@ MultiExecutorType
 JobExecutorType(DistributedPlan *distributedPlan)
 {
```

Comment:
The role of this function has shifted somewhat from being mostly about picking the executor type to mostly about diagnostics / error checking. I wonder whether keeping this function and the switch statement in FinalizePlan still makes sense. 

I don't immediately have a better proposal, but perhaps something to consider.